src/main/java/spireMapOverhaul/zones/brokenSpace/BrokenSpaceZone.java

```diff
@@ -0,0 +1,261 @@
+
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new BrokenSpaceZone(name, width, height);
+    }
+
+    @Override
+    public void renderOnMap(SpriteBatch sb, float alpha) {
+        BrokenSpaceRenderPatch.StartFbo(sb);
+        super.renderOnMap(sb, alpha);
+        BrokenSpaceRenderPatch.StopFbo(sb);
+        if (alpha > 0) {
+            FontHelper.renderFontCentered(sb, FontHelper.menuBannerFont, name,
+                    labelX * SPACING_X + OFFSET_X, labelY * Settings.MAP_DST_Y + OFFSET_Y + DungeonMapScreen.offsetY,
+                    Color.WHITE.cpy(), 0.8f
+            );
+        }
+    }
+
+    @Override
+    public boolean generateMapArea(BetterMapGenerator.MapPlanner planner) {
+        return generateNormalArea(planner, width, height);
+    }
+
+    @Override
+    protected boolean canIncludeEarlyRows() {
+        return false;
+    }
+
+    @Override
+    protected boolean canIncludeFinalCampfireRow() {
+        return false;
+    }
+
+    @Override
+    protected boolean canIncludeTreasureRow() {
+        return false;
+    }
+
+    @Override
+    public boolean canSpawn() {
+        return AbstractDungeon.actNum >= 2;
+    }
+
+    @Override
+    public Color getColor() { //I considered changing this to a variable, but a method lets you do funky stuff like a rainbow zone that changes colors or something.
+        return color;
+    }
+
+    public AbstractCard getTrulyRandomCard(Random rng) {
+        AbstractCard.CardRarity rarity = AbstractCard.CardRarity.COMMON;
+
+        // pick RANDOM rarity
+        int roll = rng.random(7);
+        switch (roll) {
+            case 0:
+            case 1:
+                rarity = AbstractCard.CardRarity.COMMON;
+                break;
+            case 2:
+            case 3:
+                rarity = AbstractCard.CardRarity.UNCOMMON;
+                break;
+            case 4:
+            case 5:
+                rarity = AbstractCard.CardRarity.RARE;
+                break;
+            case 6:
+                return AbstractDungeon.returnRandomCurse().makeCopy();
+
+        }
+
+
+        AbstractCard c = CardLibrary.getAnyColorCard(rarity).makeCopy();
+
+        float upgradeChance = ReflectionHacks.getPrivateStatic(AbstractDungeon.class, "cardUpgradedChance");
+
+        if (c.rarity != AbstractCard.CardRarity.RARE && cardRng.randomBoolean(upgradeChance) && c.canUpgrade()) {// 1857
+            c.upgrade();// 1858
+        } else {
+            for (AbstractRelic r : AbstractDungeon.player.relics) {
+                r.onPreviewObtainCard(c);
+            }
+        }
+
+
+        return c;
+
+
+    }
+
+    @Override
+    public ArrayList<AbstractCard> getAdditionalCardReward() {
+        ArrayList<AbstractCard> cards = new ArrayList<>();
+        int amount = 3;
+
+        for (AbstractRelic r : AbstractDungeon.player.relics) {
+            amount = r.changeNumberOfCardsInReward(amount);
```

Comment:
After pulling in the most recent code, you can replace this logic with `int amount = this.getNumberOfCardsInReward();`. Like with the upgrade stuff, we made a helper function for this after seeing this PR.